Menu/Matcher/Voter/AdminVoter.php

```diff
@@ -49,9 +49,19 @@ public function matchItem(ItemInterface $item)
         $match = null;
         if ($admin instanceof AdminInterface
             && $admin->hasRoute('list') && $admin->hasAccess('list')
-            && $this->request && $this->request->get('_sonata_admin') == $admin->getCode()
+            && $this->request
         ) {
-            $match = true;
+            $requestAdmin = $admin->getConfigurationPool()->getAdminByAdminCode($this->request->get('_sonata_admin'));
+
+            if ($requestAdmin instanceof AdminInterface) {
+                if ($requestAdmin->getCode() === $admin->getCode()) {
+                    $match = true;
+                }
+
+                if ($requestAdmin->isChild() && $requestAdmin->getParent()->getCode() === $admin->getCode()) {
+                    $match = true;
+                }
```

Comment:
This will be probably simplified on https://github.com/sonata-project/SonataAdminBundle/pull/4542 by something like:

```
if ($requestAdmin instanceof AdminInterface
    && $requestAdmin->getRoot()->getCode() === $admin->getCode()
) {
    $match = true;
}
```

Where `AbstractAdmin::getRoot()` returns the root parent if the admin is a child, or itself if it is not a child.